planning/behavior_velocity_intersection_module/src/scene_intersection.cpp

```diff
@@ -999,11 +1004,22 @@ bool IntersectionModule::checkCollision(
         const double end_arc_length = std::max(
           0.0, closest_arc_coords.length + (*end_time_distance_itr).second + base_link2front -
                  distance_until_intersection);
-        const auto trimmed_ego_polygon =
-          getPolygonFromArcLength(ego_lane_with_next_lane, start_arc_length, end_arc_length);
+        long double lanes_length = 0.0;
+        std::vector<lanelet::ConstLanelet> ego_lane_with_next_lanes;
+
+        for (const auto & lane : lanelets_on_path) {
+          lanes_length += bg::length(lane.centerline());
+          ego_lane_with_next_lanes.push_back(lane);
+          if (lanes_length > start_arc_length && lanes_length < end_arc_length) {
+            break;
+          }
+        }
+        const auto combined_lanelet =
+          lanelet::utils::combineLaneletsShape(ego_lane_with_next_lanes);
+        const auto combined_polygon = combined_lanelet.polygon3d();
```

Comment:
@beyzanurkaya The size of the ego_polygon is critical for collision checking with NPCs, so can you add
```
const auto trimmed_ego_polygon = getPolygonFromArcLength(ego_lane_with_next_lanes, start_arc_length, end_arc_length);
```
? This should ensure that the trimmed polygon is not empty. The rest part looks good to me.

Reply:
@soblin The approach I add didn't use `trimmed_ego_polygon`. After adding the line you marked;
```
const auto trimmed_ego_polygon = getPolygonFromArcLength(ego_lane_with_next_lanes, start_arc_length, end_arc_length);
```
 I don't quite understand where you want to use it. Can you give me more details?

Reply:
My suggestion is to resurrect `trimmed_ego_polygon` variable with this one which is calculated from `ego_lane_with_next_lanes`.

Reply:
@soblin I made the changes you wanted and also I cleaned the code. i think everything is okay now.